### ISSUE TITLE
Metric names not prefixed properly

### ISSUE THREAD

**Issue by ThomWright** [milvus-io/milvus#3176]
**Describe the bug**
Metric names are not properly prefixed with the application name.

**Steps/Code to reproduce behavior**
Use the Prometheus metrics.

**Expected behavior**
All metric names to be prefixed with `milvus_`, so it's possible to actually find them among all our other metrics.

**Environment details**
- Milvus version 0.10.1

**Additional context**
Prometheus docs about naming metrics: https://prometheus.io/docs/practices/naming/

**Comment by fishpenguin** [comment]
It's a good suggestion, we will prefix the name in 0.11.0.

**Comment by ThomWright** [comment]
In fact, please read the [Prometheus documentation on metric naming](https://prometheus.io/docs/practices/naming/). It seems clear this hasn't been read, there are many basic errors/oddities with the metrics.

Let's look at some examples.

# Example 1

`query_index_throughtout_per_microsecond`

There are at least 3 things wrong with this:

- no prefix
- not using SI unit (microsecond vs second)
- spelling (throughput, not throughtout)

Also, having a 'per_microsecond' metric seems... strange? Why not use a histogram and let the user do a `rate()` query? Is there something I'm not understanding here? I've seen a lot of Prometheus metrics, but never something like this.

# Example 2

`# HELP query_vector_response_per_microsecond the number of vectors can be queried every second `

Again, is there a reason for the `per_microsecond` metrics?

The metric name is: `query_vector_response_per_microsecond`

The help text is: `the number of vectors can be queried every second`

Either the help text or the metric name does not make any sense.

# Example 3

`CPU_temperature`

- upper casing `CPU` is a bit odd, but technically valid
- no units, is this Kelvin? Celsius? Fahrenheit?

Also, why is Milvus even reporting this metric? It's nothing to do with the Milvus application or process. It should probably be reported on by something like the [Prometheus node exporter](https://github.com/prometheus/node_exporter).

# Example 4

```
query_response_summary{quantile="0.95"} NaN
query_response_summary{quantile="0.9"} NaN
query_response_summary{quantile="0.8"} NaN
```

Is this a reason you chose to use a [Summary instead of a Histogram](https://prometheus.io/docs/practices/histograms/) here? This means I can't:

- aggregate across instances
- choose my own quantiles (e.g. 0.99)

**Comment by ThomWright** [comment]
Here are some more examples of metrics I'm unable to understand:

```
# HELP add_vectors current added vectors
# TYPE add_vectors gauge
add_vectors{instance="my-instance",job="milvus_metrics",outcome="success"} 1

# HELP add_vectors_request_total the number of vectors added
# TYPE add_vectors_request_total counter
add_vectors_request_total{instance="my-instance",job="milvus_metrics",outcome="success"} 95914
```

So we have:

- `add_vectors` Gauge
  - current added vectors
  - 1
- `add_vectors_request_total` Counter
  - the number of vectors added
  - 95914

The metric names are in an imperative tense (`add_vectors`), it would probably be better as something like `vectors_added_total` or `current_vectors_total`, depending on what it's _actually_ measuring.

Looking at the names, I would maybe guess that:

- `add_vectors` was the number of vectors stored in the system
- `add_vectors_request_total` was the number of requests which have been made to insert vectors

But, looking at the numbers, I would then conclude that I have made 95,914 requests to insert... one (1) vector. I don't _think_ that's the case. Maybe it is.

The `HELP` text is... unhelpful. It actually is making it more difficult to understand what these metrics represent. The help texts make it sound like they are the same metric. But the numbers are wildly different.

**Comment by fishpenguin** [comment]
The metric name has already been refactored as you said in master-branch. And will be released in milvus-0.12.0
